paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/factories/FlinkFactoryUtil.java

```diff
@@ -0,0 +1,544 @@
+
+import static org.apache.flink.configuration.ConfigurationUtils.canBePrefixMap;
+import static org.apache.flink.configuration.ConfigurationUtils.filterPrefixMapKey;
+import static org.apache.flink.table.factories.ManagedTableFactory.DEFAULT_IDENTIFIER;
+
+/** Utility for working with {@link Factory}s. */
+public final class FlinkFactoryUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FlinkFactoryUtil.class);
+
+    /**
+     * Describes the property version. This can be used for backwards compatibility in case the
+     * property format changes.
+     */
+    public static final ConfigOption<Integer> PROPERTY_VERSION =
+            ConfigOptions.key("property-version")
+                    .intType()
+                    .defaultValue(1)
+                    .withDescription(
+                            "Version of the overall property design. This option is meant for future backwards compatibility.");
+
+    public static final ConfigOption<String> CONNECTOR =
+            ConfigOptions.key("connector")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Uniquely identifies the connector of a dynamic table that is used for accessing data in "
+                                    + "an external system. Its value is used during table source and table sink discovery.");
+
+    public static final ConfigOption<String> FORMAT =
+            ConfigOptions.key("format")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Defines the format identifier for encoding data. "
+                                    + "The identifier is used to discover a suitable format factory.");
+
+    /**
+     * Suffix for keys of {@link ConfigOption} in case a connector requires multiple formats (e.g.
+     * for both key and value).
+     *
+     * <p>See {@link #createFlinkTableFactoryHelper(LogStoreTableFactory, Context)} Context)} for
+     * more information.
+     */
+    public static final String FORMAT_SUFFIX = ".format";
+
+    /**
+     * Creates a utility that helps in discovering formats, merging options with {@link
+     * DynamicTableFactory.Context#getEnrichmentOptions()} and validating them all for a {@link
+     * LogStoreTableFactory}.
+     *
+     * <p>The following example sketches the usage:
+     *
+     * <pre>{@code
+     * // in createDynamicTableSource()
+     * helper = FlinkFactoryUtil.createFlinkTableFactoryHelper(this, context);
+     *
+     * keyFormat = helper.discoverDecodingFormat(DeserializationFormatFactory.class, KEY_FORMAT);
+     * valueFormat = helper.discoverDecodingFormat(DeserializationFormatFactory.class, VALUE_FORMAT);
+     *
+     * helper.validate();
+     *
+     * ... // construct connector with discovered formats
+     * }</pre>
+     *
+     * <p>Note: The format option parameter of {@link
```

Comment:
We don't need such complicated documentation.

Reply:
i have removd